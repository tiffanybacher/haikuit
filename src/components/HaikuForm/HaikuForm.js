import React, { Component } from 'react';
import * as actions from '../../actions';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';

export class HaikuForm extends Component {
  state = {
    title: this.props.title || '',
    line1: this.props.line1 || '',
    line2: this.props.line2 || '',
    line3: this.props.line3 || ''
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      [name]: value
    });
  }

  handleSubmit = (e) => {
    e.preventDefault();

    const id = Date.now();

    this.props.addHaiku({ ...this.state, id });

    this.props.redirect(`/haiku/${id}`);
  }

  render() {
    return (
      <form className="HaikuForm" onSubmit={this.handleSubmit}>
        <label htmlFor="title-input">Title</label>
        <input 
          type="text" 
          id="title-input"
          name="title"
          value={this.state.title}
          onChange={this.handleChange} />
        <label htmlFor="line-1">Line 1 - 5 syllables</label>
        <input 
          type="text" 
          id="line-1-input"
          name="line1"
          value={this.state.line1}
          onChange={this.handleChange} />
        <label htmlFor="line-2">Line 2 - 7 syllables</label>
        <input 
          type="text" 
          id="line-2-input"
          name="line2"
          value={this.state.line2}
          onChange={this.handleChange} />
        <label htmlFor="line-3">Line 3 - 5 syllables</label>
        <input 
          type="text" 
          id="line-3-input"
          name="line3"
          value={this.state.line3}
          onChange={this.handleChange} />
         <button 
           className="HaikuForm-check">
           Check Syllables
        </button>
        <button 
          className="HaikuForm-submit">
          Submit
       </button>
      </form>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addHaiku: (haiku) => dispatch(actions.addHaiku(haiku)) 
});

export default connect(undefined, mapDispatchToProps)(HaikuForm);

HaikuForm.propTypes = {
  title: PropTypes.string,
  line1: PropTypes.string,
  line2: PropTypes.string,
  line3: PropTypes.string
}