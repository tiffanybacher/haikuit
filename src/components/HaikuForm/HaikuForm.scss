.HaikuForm {
  color: #CCC;
  display: flex;
  flex-direction: column;
  font-weight: 600;
  margin: 10px 0 0 0;

  input {
    border: 1px solid #DDD;
    text-align: center;
    margin: 5px 0 10px 0;
    padding: 9px;

    &::placeholder {
      color: #CCC;
    }
  }

  button {
    background-color: $purple;
    color: white;
    font-weight: 600;
    margin: 10px 0;
    padding: 10px;
  }

  .HaikuForm-submit {
    margin-bottom: 20px;
  }
}

@media only screen and (min-width: 768px) {
  .HaikuForm {
    margin: 20px 0 0 0;

    input { 
      margin: 5px 0 15px 0;
    }
  }
}