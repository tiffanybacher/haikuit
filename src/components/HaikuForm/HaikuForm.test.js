import React from 'react';
import { shallow } from 'enzyme';
import HaikuForm from './HaikuForm';

describe('HaikuForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<HaikuForm />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on change in any input', () => {
    const mockEvent = { 
      target: { name: 'title', value: 'Haiku' }
    };

    expect(wrapper.state('title')).toEqual('');

    wrapper.find('#title-input').simulate('change', mockEvent);

    expect(wrapper.state('title')).toEqual('Haiku');
  });
});