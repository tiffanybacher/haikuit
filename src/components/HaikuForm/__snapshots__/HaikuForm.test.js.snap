// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HaikuForm should match snapshot 1`] = `
<form
  className="HaikuForm"
  onSubmit={[Function]}
>
  <label
    htmlFor="title-input"
  >
    Title
  </label>
  <input
    id="title-input"
    name="title"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="line-1"
  >
    Line 1 - 5 syllables
  </label>
  <input
    id="line-1-input"
    name="line1"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="line-2"
  >
    Line 2 - 7 syllables
  </label>
  <input
    id="line-2-input"
    name="line2"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="line-3"
  >
    Line 3 - 5 syllables
  </label>
  <input
    id="line-3-input"
    name="line3"
    onChange={[Function]}
    type="text"
  />
  <button
    className="HaikuForm-submit"
  >
    Submit
  </button>
</form>
`;
